core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutor.java

```diff
@@ -0,0 +1,288 @@
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * Provides an executor interface which is used for potentially long-running tasks which may block I/O threads.
+ */
+public interface BlockingTaskExecutor extends ScheduledExecutorService {
+
+    /**
+     * Returns a {@link BlockingTaskExecutor} with a 60s timeout and unbounded work queue.
+     */
+    static BlockingTaskExecutor of() {
+        return BlockingTaskExecutorBuilder.of()
+                                          .build();
+    }
+
+    /**
+     * Returns a newly created {@link BlockingTaskExecutorBuilder}.
+     */
+    static BlockingTaskExecutorBuilder builder() {
+        return BlockingTaskExecutorBuilder.of();
+    }
+
+    /**
+     * Provides a builder that is useful for creating an {@link ScheduledExecutorService}.
+     *
+     * @see CommonPools#blockingTaskExecutor()
+     */
+    final class BlockingTaskExecutorBuilder {
+
+        private String threadNamePrefix = "armeria-blocking-tasks";
+        private int numThreads = Flags.numCommonBlockingTaskThreads();
+        private long keepAliveTime = 60;
+        private boolean allowCoreThreadTimeout = true;
+        private boolean daemon = true;
+        private int priority = Thread.NORM_PRIORITY;
+        private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
+
+        private BlockingTaskExecutorBuilder() {}
+
+        /**
+         * Returns a new builder which builds a new blocking task {@link ScheduledExecutorService}.
+         */
+        static BlockingTaskExecutorBuilder of() {
+            return new BlockingTaskExecutorBuilder();
+        }
+
```

Comment:
This isn't necessary. We can just do `new BlockingTaskExecutorBuilder()` in `BlockingTaskExecutor.builder()`.